src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,282 @@
+	bool isNull = true;
+	Form_pg_authid role = ((Form_pg_authid) GETSTRUCT(tuple));
+	AlterRoleStmt *stmt = makeNode(AlterRoleStmt);
+	const char *rolename = NameStr(role->rolname);
+
+	stmt->role = makeNode(RoleSpec);
+	stmt->role->roletype = ROLESPEC_CSTRING;
+	stmt->role->location = -1;
+	stmt->role->rolename = pstrdup(rolename);
+	stmt->action = 1;
+	stmt->options = NIL;
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("superuser", role->rolsuper));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("createdb", role->rolcreatedb));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("createrole", role->rolcreaterole));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("inherit", role->rolinherit));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("canlogin", role->rolcanlogin));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("isreplication", role->rolreplication));
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("bypassrls", role->rolbypassrls));
+
+
+	stmt->options =
+		lappend(stmt->options,
+				makeDefElemInt("connectionlimit", role->rolconnlimit));
+
+
+	rolPasswordDatum = heap_getattr(tuple, Anum_pg_authid_rolpassword,
+									pgAuthIdDescription, &isNull);
+	if (!isNull)
+	{
+		rolPassword = pstrdup(TextDatumGetCString(rolPasswordDatum));
+		stmt->options = lappend(stmt->options, makeDefElem("password",
+														   (Node *) makeString(
+															   rolPassword),
+														   -1));
+	}
+	else
+	{
+		stmt->options = lappend(stmt->options, makeDefElem("password", NULL, -1));
+	}
+
+	rolValidUntilDatum = heap_getattr(tuple, Anum_pg_authid_rolvaliduntil,
+									  pgAuthIdDescription, &isNull);
+	if (!isNull)
+	{
+		rolValidUntil = pstrdup((char *) timestamptz_to_str(rolValidUntilDatum));
+	}
+
+	stmt->options = lappend(stmt->options, makeDefElem("validUntil",
+													   (Node *) makeString(rolValidUntil),
+													   -1));
+
+	return CreateAlterRoleIfExistsCommand(stmt);
+}
+
+
+/*
+ * GenerateAlterRoleIfExistsCommandAllRoles creates ALTER ROLE commands
+ * that copies all roles from the pg_authid table.
+ */
+List *
+GenerateAlterRoleIfExistsCommandAllRoles()
+{
+	Relation pgAuthId = heap_open(AuthIdRelationId, AccessShareLock);
+	TupleDesc pgAuthIdDescription = RelationGetDescr(pgAuthId);
+	HeapTuple tuple = NULL;
+	List *commands = NIL;
+	const char *alterRoleQuery = NULL;
+
+#if PG_VERSION_NUM >= 120000
+	TableScanDesc scan = table_beginscan_catalog(pgAuthId, 0, NULL);
+#else
+	HeapScanDesc scan = heap_beginscan_catalog(pgAuthId, 0, NULL);
+#endif
+
+	while ((tuple = heap_getnext(scan, ForwardScanDirection)) != NULL)
+	{
+		const char *rolename = NameStr(((Form_pg_authid) GETSTRUCT(tuple))->rolname);
+
+		/*
+		 * The default roles and "postgres" is skipped, because reserved roles
+		 * cannot be altered.
+		 */
+		if (IsReservedName(rolename) ||
+			strcmp(rolename, "postgres") == 0)
```

Comment:
I'm still not convinced that we need anything special for `postgres` user. All the tests pass even if I remove that check. So, what's the reason we added? If you don't have a good reason, can you remove the check?